functional-test/src/main/java/org/zanata/page/AbstractPage.java

```diff
@@ -202,32 +206,89 @@ protected int getExpectedBackgroundRequests() {
         return 0;
     }
 
+
+    public void execAndWaitForNewPage(Runnable runnable) {
+        final WebElement oldPage = driver.findElement(By.tagName("html"));
+        runnable.run();
+        String msg = "new page load";
+        logWaiting(msg);
+        waitForAMoment().withMessage(msg).until(
+                new Predicate<WebDriver>() {
+                    @Override
+                    public boolean apply(WebDriver input) {
+                        try {
+                            // ignore result
+                            oldPage.getAttribute("class");
+                            // if we get here, the old page is still there
+                            return false;
+                        } catch (StaleElementReferenceException e) {
+                            // This exception means the new page has loaded
+                            // (or started to).
+                            String script = "return document.readyState === " +
+                                    "'complete' && window.javascriptFinished";
+                            Boolean documentComplete =
+                                    (Boolean) getExecutor().executeScript(
+                                            script);
+                            // TODO wait for ajax?
+                            // NB documentComplete might be null/undefined
+                            return documentComplete == Boolean.TRUE;
```

Comment:
Adapted from http://www.obeythetestinggoat.com/how-to-get-selenium-to-wait-for-page-load-after-a-click.html